src/utop.ml

```diff
@@ -57,7 +57,12 @@ let setup sctx ~dir =
   let expander = Super_context.expander sctx ~dir in
   let scope = Super_context.find_scope_by_dir sctx dir in
   let db = Scope.libs scope in
-  let libs = libs_under_dir sctx ~db ~dir:(Path.build dir) in
+  let libs = libs_under_dir sctx ~db ~dir:(Path.build dir)
+    |> List.filter ~f:(fun (lib : Lib.t) ->
```

Comment:
How about we just do this filtering in `libs_under_dir`? This function is only used once so it makes sense not to require the caller to post-process further. Also, it would deforest things a little bit by avoiding the intermediate list.

Reply:
Yes indeed, fixed that while rebasing.